workflowproj/workflowproj_test.go

```diff
@@ -112,10 +113,21 @@ func Test_Handler_WorkflowMinimalAndPropsAndSpecAndGeneric(t *testing.T) {
 	assert.Equal(t, "02-hello-resources", proj.Resources[1].Name)
 	assert.Equal(t, proj.Workflow.Spec.Resources.ConfigMaps[0].ConfigMap.Name, proj.Resources[0].Name)
 	assert.Equal(t, proj.Workflow.Spec.Resources.ConfigMaps[1].ConfigMap.Name, proj.Resources[1].Name)
-	assert.NotEmpty(t, proj.Resources[0].Data, fmt.Sprintf("Data in proj.Resources[0] is empty %+v", proj.Resources[0]))
-	assert.NotEmpty(t, proj.Resources[1].Data, fmt.Sprintf("Data in proj.Resources[1] is empty %+v", proj.Resources[1]))
-	assert.NotEmpty(t, proj.Resources[0].Data["myopenapi.json"])
-	assert.NotEmpty(t, proj.Resources[1].Data["input.json"])
+	data, err := getResourceDataWithFileName(proj.Resources, "myopenapi.json")
```

Comment:
We shouldn't need this since the array is sorted. But I guess we can force it here to avoid inconclusive tests all the time. Weirdly, I wasn't able to reproduce this error locally.

Reply:
I was able to reproduce the failure locally and it was just as you said in slack: the order of the resources, being a slice, is random and nondeterministic. In practice the order doesn't matter when executing the workflow, but it impacts the unit test, so adding this function to retrieve the contents should suffice without impacting the functional code.

Reply:
So I guess the slice order we are doing before is not working. :/